src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRequest.cs

```diff
@@ -191,6 +201,38 @@ public CertificateRequest(X500DistinguishedName subjectName, PublicKey publicKey
             HashAlgorithm = hashAlgorithm;
         }
 
+        /// <summary>
+        ///   Create a CertificateRequest for the specified subject name, encoded public key, hash algorithm,
+        ///   and RSA signature padding.
+        /// </summary>
+        /// <param name="subjectName">
+        ///   The parsed representation of the subject name for the certificate or certificate request.
+        /// </param>
+        /// <param name="publicKey">
+        ///   The encoded representation of the public key to include in the certificate or certificate request.
+        /// </param>
+        /// <param name="hashAlgorithm">
+        ///   The hash algorithm to use when signing the certificate or certificate request.
+        /// </param>
+        /// <param name="rsaSignaturePadding">
+        ///   The RSA signature padding to use when signing this request with an RSA certificate.
+        /// </param>
+        public CertificateRequest(
+            X500DistinguishedName subjectName,
+            PublicKey publicKey,
+            HashAlgorithmName hashAlgorithm,
+            RSASignaturePadding? rsaSignaturePadding = null)
+        {
+            ArgumentNullException.ThrowIfNull(subjectName);
+            ArgumentNullException.ThrowIfNull(publicKey);
+            ArgumentException.ThrowIfNullOrEmpty(hashAlgorithm.Name, nameof(hashAlgorithm));
+
+            SubjectName = subjectName;
+            PublicKey = publicKey;
+            HashAlgorithm = hashAlgorithm;
+            _rsaPadding = rsaSignaturePadding;
```

Comment:
This gets lazily checked in `Create`. Would it be helpful to throw an argument exception during the ctor if `publicKey.Oid == Oids.Rsa`?

Reply:
It would when it applies, but the generator-accepting overloads don't use it, so checking too early is over-zealous.